### ISSUE TITLE
Sizing of EMF files incorrect and unchanging.

### ISSUE THREAD

**Issue by parchd-1** [jgm/pandoc#3798]
Images in EMF format exported to docx do not have the correct size or aspect ratio, and it does not seem possible to set the size. This was previously reported in #2720 but the reporter closed the issue before it was resolved.

Example markdown document (also in attached zip with emf and output docx [problem.zip](https://github.com/jgm/pandoc/files/1146421/problem.zip)):

```markdown
No width given:

![testimg](./squarecircle.emf)

With percentage:

![2testimg](./squarecircle.emf){width=100%}

With units:

![3testimg](./squarecircle.emf){#bah width=6cm}

With units:

![4testimg](./squarecircle.emf){width=6in height=3in}

```

The size / proportions of the image seem to be extractable from the file:
From the attached file, opening it with less I see "Screen=10205x13181px, 216x279mm" and "Drawing=576.0x288.0px, 152.4x76.2mm"

**Comment by jgm** [comment]
We'd need to add support for EMF to Text.Pandoc.ImageSize.
It seems that the size information is fairly easy to extract, so that's good.

**Comment by parchd-1** [comment]
I have also found https://stackoverflow.com/questions/3391087/extracting-width-height-dimensions-from-emf-image-files#3391180 which links to https://msdn.microsoft.com/en-us/library/cc230514.aspx. They might also be helpful.

**Comment by parchd-1** [comment]
I had a little look at Text.Pandoc.ImageSize and the docx writer, so I think the answer to this is probably no, but would this be possible to do using a filter
 I'm a bit desperate for it, although I will use high-dpi png files in an emergency this would obviously be much better. Filters I can cope with, especially thanks to panflute, but haskell not so much.

**Comment by mb21** [comment]
> I had a little look at Text.Pandoc.ImageSize and the docx writer, so I think the answer to this is probably no

the answer to what?

you can certainly do it in a filter as well, but it would be preferably to have this in pandoc itself in the long run...

> but haskell not so much

see e.g. http://learnyouahaskell.com :)

**Comment by parchd-1** [comment]
> the answer to what?
the answer to the whether it was possible in a filter. Sadly I didn't accomplish it, so in the meantime I'll have to go with massive file sizes from the pngs.

**Comment by parchd-1** [comment]
I did work out how to do this in a filter in the end -  I was having a dense moment.

Not posting an actual filter here because it ended up being part of a bigger filter specific for this document, but this python script for the command line returns either the dimensions in unknown units, or returns dimensions based on a user-given width.

As I note, there seem to be two slightly different sets of dimensions in the file. Without digging deeper, can't see which is "correct".

```python
#!/usr/bin/env python3

from sys import argv

fd = open(argv[1], 'rb')

if len(argv) > 2:
    want_width = float(argv[2])
else:
    want_width = None

fd.seek(8)

top = int.from_bytes(fd.read(4), 'little')
left = int.from_bytes(fd.read(4), 'little')
bottom = int.from_bytes(fd.read(4), 'little')
right = int.from_bytes(fd.read(4), 'little')

# and then another 4 with similar but not identical ratios?

# top = int.from_bytes(fd.read(4), 'little')
# left = int.from_bytes(fd.read(4), 'little')
# bottom = int.from_bytes(fd.read(4), 'little')
# right = int.from_bytes(fd.read(4), 'little')

height = bottom - top
width = right - left

if want_width:
    ratio = width / height
    width = want_width
    height = width * ratio

print("{} {}".format(width, height))
```

**Comment by agusmba** [comment]
Regarding the "unchanging" part of the issue, I experienced the same today, when inserting the same image multiple times with different widths (md -> docx). Only the first one is calculated, and the rest appear with the same size independently of that is configured in brackets.

Should I create a different issue or is this one good enough?

**Comment by mb21** [comment]
@agusmba that indeed sounds like a seperate issue. can you post the exact input with pandoc, word version etc. in a seperate issue?

**Comment by andymac-2** [comment]
@mb21 I could try to add EMF support, I've read through the spec and it seems straightforwards enough. However, the EMF file supplied by @parchd-1 , and all EMF files produced by Inkscape, have mangled headers and won't be recognised. This appears to a bug with Inkscape.

**Comment by jgm** [comment]
closed by #4375
